services/src/util/retry.rs

```diff
@@ -0,0 +1,43 @@
+use std::time::Duration;
+
+use futures::Future;
+
+/// A method wrapper for calling a method that may fail spuriously until it succeeds.
+/// The method is called at most `max_retries` times.
+/// If it still fails after `max_retries` times, the error is returned.
+///
+/// Uses exponential backoff by taking the `initial_delay_ms` and multiplying each time an `exponential_backoff_factor` on it.
+///
+/// # Panics
+/// Panics if `max_retries` is 0.
+///
+pub async fn retry<F, T, E, Fut>(
```

Comment:
can/should we have a test just for this method?

Reply:
9c9ef30